swap_meet/vendor.py

```diff
@@ -1,2 +1,219 @@
-class Vendor:
-    pass
\ No newline at end of file
+from swap_meet.item import Item
+
+# Wave 1- SJ
+# 1. created the Vendor class
+# 2. one attribute named inventory, empty list by default
+
+class Vendor: 
+    
+    def __init__(self, inventory=None):
+        self.inventory =  inventory if inventory is not None else []
+    
+    
+    # Wave 1- SJ
+    # 3. Created instance method named add, which takes in one item
+    # 3a. This method adds an item to the inventory
+    # 3b. This method returns the item that was added
+
+    def add(self, added_item):
+        self.added_item = added_item
+            
+        self.inventory.append(added_item)
+        return added_item
+        
+        
+    # Wave 1- SJ
+    # 4. Created instance method named remove, which takes in one item
+    # 4a. If a matching item is found, removes the item from the inventory
+    # 4b. If no matching item is found, returns False
+
+    def remove(self, removed_item):
+        self.removed_item = removed_item
+            
+        if removed_item in self.inventory: 
+            self.inventory.remove(removed_item)
+            return removed_item
+        else:
+            return False
+        
+        
+    # Wave 2- SJ
+    # 5. Created an instance method that takes one argument, id an integer, 
+    #    id represents an Item's id number in the inventory
+    # 5a. This method returns the item from the inventory with the matching id 
+    # 5b. If no match is found, the method returns None 
+    
+    def get_by_id(self, id):
+        
+        for item in self.inventory:
+            if item.id == id:
+                return item
+            
+        return None
+    
+    
+    # Wave 3- KV
+    # 6. Created instance method named swap_items, takes 3 arguments
+    # other_vendor, my_item, their item
+    # 6a. if my item isn't in my inventory, OR if their item isn't in their inventory,
+    # 6b. the return False
+    # 6c. Otherwise, remove the item from my inventory
+    # 6d. Add that item to their inventory 
+    # 6e. Likewise, if the other vendors item is in their inventory, remove that item
+    # 6f. Add that item to my inventory and retrun True
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
```

Comment:
An optional enhancement could be to check if either vendor has the item that they are receiving in the swap inside of their inventory already.

Reply:
whooooole 'nother level, lol